### ISSUE TITLE
Mac shortcut not working on Release Note Screen

### ISSUE THREAD

**Issue by rkdmen** [Microsoft/vscode#71800]
- VSCode Version: 1.33.0 ( Recent )
- OS Version: Mac 10.14.4

Steps to Reproduce:

1. On Release Note Screen
2. (hide or minimize) `cmd+h` or `cmd+m` does not work

Does this issue occur when all extensions are disabled?: Yes

**Comment by rkdmen** [comment]
Also yank back killed text `ctrl+y` stopped working.
Works on all other program except vscode.

**Comment by timdmackey** [comment]
I'm having this issue everywhere in VSCode, regardless of the view or document open. Issue still occurs when all extensions are disabled.

VSCode Version: 1.56.2
MacOS 10.15.7

**Comment by douira** [comment]
I also have this issue. I can't even hide the window by right clicking on the application icon in the dock. When I try to hide it that way, the window hides for a split second and then re-appears immediately. cmd + H only works and actually hides the window when a native dialog like the "about" window or the dialog that opens if you close an unsaved file.
```
Version: 1.57.0
Commit: b4c1bd0a9b03c749ea011b06c6d2676c8091a70c
Date: 2021-06-09T17:22:31.215Z
Electron: 12.0.9
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 19.6.0
```

**Comment by jacobweber** [comment]
Same here. For example, open Git Graph and close it. From then on, while that workspace is open, pressing command-H on the Mac will no longer hide the app.

**Comment by ryanromanov** [comment]
I'm unable to hide or minimize with cmd + h / cmd + m regardless of where I am in the editor. 

VS Code Version: 1.58.2.

**Comment by babramczyk** [comment]
Moved here, for a better title and visibility:

https://github.com/microsoft/vscode/issues/130194

**Comment by lramos15** [comment]
@mjbvz I think this issue is a little bit more annoying than initially because it breaks the keybinding for the entire VS code session until you quit the window and re-open it.

**Comment by babramczyk** [comment]
@rkdmen could we rename the issue to clarify that these shortcuts are not just broken during webviews, but rather always? Just want to make sure it gets the attention it deserves 🙂

**Comment by rolf-langer** [comment]
Happens with current latest version 1.59.0 under macOS Catalina (10.15.7) just with normal code views. Maybe related to the display of the changelog after update. App restart does fix this behavior for me.

**Comment by nebbles** [comment]
I was seeing this on the following version:

> Version: 1.58.2
> Commit: c3f126316369cd610563c75b1b1725e0679adfb3
> Date: 2021-07-14T22:09:06.581Z
> Electron: 12.0.13
> Chrome: 89.0.4389.128
> Node.js: 14.16.0
> V8: 8.9.255.25-electron.0
> OS: Darwin x64 19.6.0

After updating to 1.59.0 seems to have fixed it (although maybe that was just the required restart when updating).

**Comment by babramczyk** [comment]
I’m on `1.59.0`, still broken for me

**Comment by operatorone** [comment]
Version: 1.59.1
Commit: 3866c3553be8b268c8a7f8c0482c0c0177aa8bfa
Date: 2021-08-19T11:53:52.479Z
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
V8: 9.1.269.36-electron.0
OS: Darwin x64 20.5.0

Also broken for me.

**Comment by mhutchie** [comment]
Confirmed that the bug @jacobweber referenced above almost 3 months ago is still occurring in Visual Studio Code 1.61.1.

Version: 1.60.1
Commit: 83bd43bc519d15e50c4272c6cf5c1479df196a4d
Date: 2021-09-10T17:09:14.403Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

@mjbvz Would you please consider increasing the priority of this issue, as it has been affecting users for quite some time (worsened by the Webview changes a few months back)?

**Comment by babramczyk** [comment]
@rkdmen could we rename the issue to clarify that these shortcuts are not just broken during webviews, but rather always? Just want to make sure it gets the attention it deserves 🙂 (and the VS Code team closed my other issue, that had a different title)

**Comment by operatorone** [comment]
Vielen Dank für Ihre E-Mail.

Die medienflieger sind unterwegs und ab dem 05.10.2021 wieder zu erreichen.
Ihre E-Mail wird erst nach unserer Rückkehr bearbeitet. 

Viele Grüße
Tilman Behrend

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Tilman Behrend | medienflieger GmbH | An der Sicker 7 | 97318 Kitzingen
t: 09321.388 28 10 | f: 09321.388 28 09
www.medienflieger.de | ***@***.***

**Comment by rkdmen** [comment]
Confirmed still not working with latest OS/Vscode. Updated in detail.

**Comment by rantler** [comment]
Still experiencing this on my machine. It is intermittent though and only happens after VSCode has been running for some time. If I quit and restart the VSCode, the Cmd-H works as expected for some period of time afterward.

```
Version: 1.60.2
Commit: 7f6ab5485bbc008386c4386d08766667e155244e
Date: 2021-09-22T11:59:27.195Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0
```

**Comment by iammorjj** [comment]
@guimaraes13 I'm surprised, that after reloading VSCode shortcut started working! TY!

**Comment by egorgam** [comment]
Doesn't work for me too

Version: 1.62.1
Commit: f4af3cbf5a99787542e2a30fe1fd37cd644cc31f
Date: 2021-11-05T09:23:14.144Z (4 days ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.1.0

**Comment by malhar-trivedi** [comment]
For me cmd + h works after launching the vscode but after few hours, it stops working. I have to restart to make it work again.

Version: 1.62.1
Commit: f4af3cbf5a99787542e2a30fe1fd37cd644cc31f
Date: 2021-11-05T09:23:14.144Z (6 days ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 20.6.0

**Comment by Fischmuetze** [comment]
same here 

> For me cmd + h works after launching the vscode but after few hours, it stops working. I have to restart to make it work again.

Version: 1.62.2
Commit: 3a6960b964327f0e3882ce18fcebd07ed191b316
Datum: 2021-11-11T20:59:05.913Z (Vor 3 Tagen)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
Betriebssystem: Darwin x64 20.6.0

**Comment by ilyabirman** [comment]
Same problem for me in 1.62.2, annoying as hell :–( I had no idea I used to press ⌘H every minute or so. Please please fix!

**Comment by crh19970307** [comment]
Command-M still not working for me!!!!!!
My environment:
VsCode Version: 1.62.2
MacOS Version: 10.15.5

**Comment by kdzhang** [comment]
same here

> For me cmd + h works after launching the vscode but after few hours, it stops working. I have to restart to make it work again.

Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T08:14:33.711Z (1 wk ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 21.1.0

**Comment by icezohu** [comment]
![image](https://user-images.githubusercontent.com/3282625/145136224-10b3dbde-37ff-43b7-9bc7-4eca1c4c26f7.png)
![image](https://user-images.githubusercontent.com/3282625/145136258-575138b8-4320-4ae6-8755-81a6eb32cc66.png)
changing these key bindings could fix the problem

Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.1.0

**Comment by Fischmuetze** [comment]
yes, thank you! the first one was helping for my configuration - solved!

**Comment by andrewmcdonald-oxb** [comment]
Made no difference to me unfortunately. Not sure I remember that "toggle test history in peek" command being there when I first started experiencing the issue. It's still a problem after viewing markdown preview in the VS Code window, such as the release notes.

**Comment by ra4ul** [comment]
This broken Cmd+h binding has existed for years now.  It manifests (at least on my Mac) only after opening certain kinds of files.  One sure fire way to reproduce it is to open up the Preview version of a Markdown file (with Cmd+Shift+v) and then close the Preview.  Cmd+h no longer functions while that window is active, but will work properly if you open up a new window.

Is this really not a priority given its importance to accessibility?

**Comment by angusm73** [comment]
Success! ✅
It's working for me after:

- Remove key binding for `Toggle test history in peek`
- Restart vscode
- `cmd+h` & `cmd+m` work now! 🎉

Would be nice if vscode didn't overwrite this global keyboard shortcut though.

**Comment by bengry** [comment]
> Success! ✅ It's working for me after:
> 
> * Remove key binding for `Toggle test history in peek`
> * Restart vscode
> * `cmd+h` & `cmd+m` work now! 🎉
> 
> Would be nice if vscode didn't overwrite this global keyboard shortcut though.

Doesn't fix it for me.

**Comment by godly-devotion** [comment]
> > Success! ✅ It's working for me after:
> > 
> > * Remove key binding for `Toggle test history in peek`
> > * Restart vscode
> > * `cmd+h` & `cmd+m` work now! 🎉
> > 
> > Would be nice if vscode didn't overwrite this global keyboard shortcut though.
> 
> Doesn't fix it for me.

Yup doesn't work for me either. It works initially because VS Code was restarted but after some time `cmd + h` no longer works again despite removing the key bindings.

Also couldn't help but notice that this bug is as old as covid..

**Comment by operatorone** [comment]
Yeah, same here. I thought the removal of the keybinding helped initially, I discovered it didn't after a short time.

**Comment by bfsmith9** [comment]
> Success! ✅ It's working for me after:
> 
> * Remove key binding for `Toggle test history in peek`
> * Restart vscode
> * `cmd+h` & `cmd+m` work now! 🎉
> 
> Would be nice if vscode didn't overwrite this global keyboard shortcut though.

Yes, this worked for me too. It also fixed "Hide others", which is cmd + option + h. Thank you.

**Comment by oliverepper** [comment]
Still not working for me. No keybindings inside VSCode.

**Comment by connor4312** [comment]
> Are these commands from the test explorer? Do they need to better scoped?

This is scoped for when the peek is open: 

https://github.com/microsoft/vscode/blob/994f907fa8b39c89440bc01b2486c7380721f786/src/vs/workbench/contrib/testing/browser/testingOutputPeek.ts#L1722

also, this issue predates testing support in VS Code, so I'm not sure it's the same.

**Comment by larzon83** [comment]
> also, this issue predates testing support in VS Code, and people report that even without the keybinding it stops working, so I'm not sure it's the root cause.

I also have no keybindings. Still stops working.

**Comment by mjbvz** [comment]
@larzon83 @oliverepper Have you tried removing the keybindings as suggested here: https://github.com/microsoft/vscode/issues/71800#issuecomment-997485090

**Comment by jokocide** [comment]
> > Success! ✅ It's working for me after:
> > 
> > * Remove key binding for `Toggle test history in peek`
> > * Restart vscode
> > * `cmd+h` & `cmd+m` work now! 🎉
> > 
> > Would be nice if vscode didn't overwrite this global keyboard shortcut though.
> 
> Doesn't fix it for me.

Worked for me, I might suggest that you reset all of your keybindings for VScode and MacOS to defaults and then erase the suggested keybinding.

**Comment by NiekPas** [comment]
> > > Success! ✅ It's working for me after:
> > > 
> > > * Remove key binding for `Toggle test history in peek`
> > > * Restart vscode
> > > * `cmd+h` & `cmd+m` work now! 🎉
> > > 
> > > Would be nice if vscode didn't overwrite this global keyboard shortcut though.
> > 
> > 
> > Doesn't fix it for me.
> 
> Worked for me, I might suggest that you reset all of your keybindings for VScode and MacOS to defaults and then erase the suggested keybinding.

@jokocide Are you sure this fixed it persistently and not just for a few hours?

**Comment by jokocide** [comment]
> > > > Success! ✅ It's working for me after:
> > > > 
> > > > * Remove key binding for `Toggle test history in peek`
> > > > * Restart vscode
> > > > * `cmd+h` & `cmd+m` work now! 🎉
> > > > 
> > > > Would be nice if vscode didn't overwrite this global keyboard shortcut though.
> > > 
> > > 
> > > Doesn't fix it for me.
> > 
> > 
> > Worked for me, I might suggest that you reset all of your keybindings for VScode and MacOS to defaults and then erase the suggested keybinding.
> 
> @jokocide Are you sure this fixed it persistently and not just for a few hours?

I noticed that it stopped working the next morning, but closing and re-opening VScode restored functionality again.

**Comment by andrewmcdonald-oxb** [comment]
For everyone noticing it starts then stops working, can we check whether this coincides with displaying a markup document (such as the new version readme which automatically appears when VSCode updates itself)?

**Comment by chalkygames123** [comment]
Ha, that seems to be the exact cause! Once I select Help > Release Notes, `cmd+h` and `alt+cmd+h` no longer work.

**Comment by larzon83** [comment]
> Ha, that seems to be the exact cause! Once I select Help > Release Notes, `cmd+h`, `alt+cmd+h` and `cmd+m` no longer work.

Yes, I can reproduce it as well!

**Comment by NiekPas** [comment]
> For everyone noticing it starts then stops working, can we check whether this coincides with displaying a markup document (such as the new version readme which automatically appears when VSCode updates itself)?

Yes, I can reproduce this as well.

**Comment by douira** [comment]
I can also reproduce. The issue persists even after closing the markdown preview. It does not, however, persist when I close the window without quitting the app.

**Comment by egorgam** [comment]
It also reproduces and VERY frustrating(

**Comment by ryanswanson** [comment]
I have reproduced this with markdown preview as well and noticed something that may help diagnose the issue.

From a markdown file (*.md), opening the markdown preview (I use a shortcut for the 'Markdown > Open Preview to the Side' command) will move **focus** to the newly opened markdown preview tab.

While the markdown preview tab is focused, Cmd+M does not function, but if I move focus back to the Editor for the markdown file (*.md), Cmd+M works again to minimize the window.

If the markdown preview tab is focused when the tab is closed, Cmd+M in that window will no longer function...until you re-open the markdown preview and move focus back out.

If the markdown preview tab is NOT focused when the tab is closed, Cmd+M in that window will continue to function properly thereafter.

It seems the markdown preview claims the key binding when focused, releases it upon blur, but does not release it upon close...or something to that effect (I'm just another user here).

I have also not been able to determine what markdown preview is doing with the Cmd+M binding (no visible effect is seen).

**Comment by NiekPas** [comment]
@ryanswanson 

> While the markdown preview tab is focused, Cmd+M does not function, **but if I move focus back to the Editor for the markdown file (*.md), Cmd+M works again to minimize the window.**

Sorry for overcomplicating the issue, but this only works for me when I move focus back to the Editor using ctrl+tab, _not_ when switching tabs by clicking on them with the mouse.

Even more strangely: moving focus back to the editor only re-enables cmd+m/cmd+h if, while switching tabs using ctrl+tab, I press ctrl+tab such that the tab switcher pops up; then **hold down the ctrl key for a short while, and only then release the ctrl key**. Only if I hold down the ctrl key for long enough, such that the tab switcher is open for a while, do the window keyboard shortcuts work again. Some very rudimentary testing using a metronome seems to indicate that the 'activation threshold' is around 200-300 ms.

Can anyone reproduce this bizarre behaviour?

Version: 1.64.2

**Comment by jwsy** [comment]
I'm able to reproduce that bizarre behavior as well, but cmd+h/cmd+m both break after I open up a Markdown preview.

Version: 1.65.2
Commit: c722ca6c7eed3d7987c0d5c3df5c45f6b15e77d1
Date: 2022-03-10T14:33:49.188Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.4.0

**Comment by ivancuric** [comment]
Same thing here

Version: 1.66.0
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:49:37.118Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin arm64 21.4.0

**Comment by pentago** [comment]
Same thing here

Version: 1.67.0-insider
Commit: 95e8ecaa65bf2d790a4e3b586e843435fc5c0f77
Date: 2022-03-31T05:28:22.904Z (6 hrs ago)
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin arm64 21.4.0

**Comment by rkdmen** [comment]
Three years after reporting bug, I'm still having this issue.

**Comment by Ryouku** [comment]
Same here.


Version: 1.66.0
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:49:37.118Z (2 days ago)
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin arm64 21.4.0

**Comment by 10bo** [comment]
Same thing here 😴

Version: 1.66.0
Commit: e18005f0f1b33c29e81d732535d8c0e47cafb0b5
Date: 2022-03-30T05:49:37.118Z (5 days ago)
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin arm64 21.1.0

**Comment by jacobweberbowery** [comment]
My workaround is to just define a Keyboard Maestro macro for command-H that hides VSCode. Not perfect, but it works.

**Comment by Trandung5698** [comment]
Vào 16:48, T.6, 1 Th4, 2022 Zy ***@***.***> đã viết:

**Comment by Trandung5698** [comment]
Vào 18:04, T.2, 4 Th4, 2022 10bo ***@***.***> đã viết:

**Comment by 10bo** [comment]
> I've been feeling the Cmd+H / Cmd+M bug for a long time, I didn't realize all this time that it was the markdown previewer! Glad we at least know where it's coming from! I wish I could disable the previewer altogether given this effect — I don't recall even using it. I usually stick to Typora..
> 
> Regardless — a note on workarounds
> 
> > Sorry for overcomplicating the issue, but this only works for me when I move focus back to the Editor using ctrl+tab, _not_ when switching tabs by clicking on them with the mouse.
> 
> I find this same behavior as well but I also realized that the "Double Click To Switch To Editor" feature also hands back focus in a way that resumes standard Cmd+H / Cmd+M control, and I believe this feature is on by default. Just open a markdown file, open the previewer, then double click anywhere in the previewer to hand focus back and get your keybindings back.
> 
> It's almost funny. With both panes open you can click on the previewer, nothing works, double click it, key-binds work again, click the previewer and...they're back off.

Thank you for pointing out this workaround. This has worked for me.

The workaround for anyone experiencing this issue with Markdown:

1. Create new file (**⌘+N** or **Ctrl+N**).
2. Change language mode to Markdown (**⌘+P** or **Ctrl+P** then select Change Language Mode).
3. Open locked preview to side (**⌘⇧+P** or **Ctrl⇧+P** then select Markdown: Open Locked Preview to the Side)
4. Double click anywhere on the locked preview.

Note: Ensure 'Markdown > Preview: Double Click To Switch' is enabled in settings.

**Comment by BB9z** [comment]
I have a workaround:

Open command palette and search: Open Keyboard Shortcuts (JSON)

Then put this item in config file:

```json
{
        "key": "",
        "command": "testing.toggleTestingPeekHistory"
}
```

Restart VSCode then it will be fine.

**Comment by ktravelet** [comment]
> I have a workaround:
> 
> Open command palette and search: Open Keyboard Shortcuts (JSON)
> 
> Then put this item in config file:
> 
> ```json
> {
>     "key": "",
>     "command": "testing.toggleTestingPeekHistory"
> }
> ```
> 
> Restart VSCode then it will be fine.

FINALLY!! thank you!

**Comment by esthor** [comment]
> I have a workaround:
> 
> Open command palette and search: Open Keyboard Shortcuts (JSON)
> 
> Then put this item in config file:
> 
> ```json
> {
>     "key": "",
>     "command": "testing.toggleTestingPeekHistory"
> }
> ```
> 
> Restart VSCode then it will be fine.

This also worked for me.

And just a note that making this change in the visual editor for Keyboard Shortcuts (Code -> Preferences -> Keyboard Shortcuts) does _not_ solve it. You really do need to go into the Keyboard Shortcuts JSON and make the change. The important thing for me was to remove the `"when": "testing.isPeekVisible"`. If I leave that, it still breaks the cmd+h/cmd+m behavior anytime I open a Markdown preview.

**Comment by Bill-Haku** [comment]
> I have a workaround:
> 
> Open command palette and search: Open Keyboard Shortcuts (JSON)
> 
> Then put this item in config file:
> 
> ```json
> {
>     "key": "",
>     "command": "testing.toggleTestingPeekHistory"
> }
> ```
> 
> Restart VSCode then it will be fine.

Thanks, it worked for me!

**Comment by NiekPas** [comment]
> I have a workaround:
> 
> Open command palette and search: Open Keyboard Shortcuts (JSON)
> 
> Then put this item in config file:
> 
> ```json
> {
>     "key": "",
>     "command": "testing.toggleTestingPeekHistory"
> }
> ```
> 
> Restart VSCode then it will be fine.

I had this item in my config, but still ran into this bug today. Restarting VS Code obviously did temporarily fix the issue, but that's not exactly a viable workaround...

**Comment by rkdmen** [comment]
> > I have a workaround:
> > Open command palette and search: Open Keyboard Shortcuts (JSON)
> > Then put this item in config file:
> > ```json
> > {
> >     "key": "",
> >     "command": "testing.toggleTestingPeekHistory"
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Restart VSCode then it will be fine.
> 
> I've had this item in my config for a while, but still ran into this bug today. Restarting VS Code obviously did temporarily fix the issue, but that's not exactly a viable workaround...

Same here, I tried over week with this shortcut, but bug still exist.

**Comment by alfuken** [comment]
How many more comments and upvotes should this issue receive to get it fixed?...

**Comment by ssigwart** [comment]
I looked into this and created a PR for it.  It does not make Cmd+H or Cmd+M work while in a webview (e.g. Markdown preview), but it does fix it breaking those keybindings until you reboot.  More details are in the PR.

As a workaround until the PR is merged, you can make sure that you focus on a different editor before closing a Markdown preview. You can also use that trick to get it working again instead of having to completely restart VS Code if you accidentally do close the editor while focused on the webview.

**Comment by timdmackey** [comment]
Thank you @ssigwart! The pull request is here for those interested in more details: https://github.com/microsoft/vscode/pull/154648

**Comment by djmtype** [comment]
> I have a workaround:
> 
> Open command palette and search: Open Keyboard Shortcuts (JSON)
> 
> Then put this item in config file:
> 
> ```json
> {
>     "key": "",
>     "command": "testing.toggleTestingPeekHistory"
> }
> ```
> 
> Restart VSCode then it will be fine.

I already had: 
```
{
  "key": "cmd+h",
  "command": "testing.toggleTestingPeekHistory",
  "when": "testing.isPeekVisible"
}
```
Is this not the same? However, this doesn't always do the trick. It's also hard to reproduce this bug because with `cmd+h` sometimes I can get VS Code to hide and other times it won't.

**Comment by CodingMarkus** [comment]
I have the issue as well but the workaround doesn't work for me as `testing.isPeekVisible` has never been set to `cmd+h` in the first place, it's set to `opt+h`. There is no shortcut set for `cmd+h` at all anywhere and still it fails for certain windows while it works for other windows. Also the window it fails has never been a markup preview, it's an editor editing a shell script and never has been anything else. Two other open windows work as expected.

**Comment by cfarvidson** [comment]
> I have the issue as well but the workaround doesn't work for me as `testing.isPeekVisible` has never been set to `cmd+h` in the first place, it's set to `opt+h`. There is no shortcut set for `cmd+h` at all anywhere and still it fails for certain windows while it works for other windows. Also the window it fails has never been a markup preview, it's an editor editing a shell script and never has been anything else. Two other open windows work as expected.

Same for me. I can't find "cmd+h" but hiding vscode does not work.

**Comment by Lawrence-of-AnKing** [comment]
me too...

I can not hide my vsc by `Command` + `H`.

And I've been annoyed by this problem for a looooong time...

**Comment by operatorone** [comment]
Mee too. I can't understand that this is consistently not solved and it stresses me every day.

**Comment by juliand665** [comment]
For what it's worth, I've recently noticed it happening only in environments that are not the regular text editor, so e.g. a Jupyter notebook. The "What's New" view for patch notes is another one, iirc.

**Comment by maximillianus** [comment]
This happens to me too especially `Cmd+M` in markdown editing using Markdown All In One plugin installed. This is very annoying as `Cmd+M` in Markdown All in One plugin is replaced into shortcut for **Markup**.

`Cmd+H` and `Cmd+M` in markdown preview does not work too.

**Comment by mocenigo** [comment]
Hello. it is still there and it is very annoying. If I am in a web view, cmd-H will not hide the application, and I am forced to click in a text pane first.

**Comment by nurp** [comment]
Somehow My Mac M1 (13.5 Ventura) doesn't hide the last window whether it is VSCode (1.84) or terminal or Chrome. Unhide another window and see if you can hide VSCode now. I think it is a mac bug.

**Comment by nbbaier** [comment]
I have been experiencing this behavior a lot lately on VSCode 1.87.2 running on a M2 mac (Sonoma 14.4). Has anyone found a consistent workaround so that that ⌘H works?

**Comment by mahboud** [comment]
I'm also seeing that ⌘H, ⌥⌘H and ⌘M haven't been working in VSCode for months.

**Comment by rkdmen** [comment]
@mjbvz Issue still happens on `Release Notes` page view.

**Comment by josh-xwm** [comment]
To fix hiding VSCode not working, download [Hammerspoon](https://www.hammerspoon.org/) and add the below to your config, then make sure you click reload config in the Hammerspoon app. Also make sure Hammerspoon has been added and given permissions in `macOS System Settings` → `Privacy & Security` → `Input Monitoring`
as well as → `Privacy & Security` → `Accessibility`

Should fix vscode not hiding with markdown previews open, I don't have issues with hiding VSCode in other situations but this will likely fix that as well.

You can likely come up with a similar solution for minimizing just ask AI to generate a hammerspoon lua script for it

```lua
--[[
  Workaround for VS Code Markdown Preview intercepting Cmd+H.
  Force hide VS Code when Cmd+H is pressed and VS Code is frontmost.
--]]

if cmdHEventTap then
    cmdHEventTap:stop()
    cmdHEventTap = nil
end

-- Define the bundle identifiers for VS Code installations
-- Verify these using `mdls -name kMDItemCFBundleIdentifier /Applications/Visual\ Studio\ Code.app` in Terminal if needed.
local vscodeBundleIDs = {
  ["com.microsoft.VSCode"] = true,          -- Stable version
  ["com.microsoft.VSCodeInsiders"] = true,  -- Insiders version
  ["com.visualstudio.code.oss"] = true      -- OSS version (if you use it)
  -- Add any other custom builds or portable versions you use here
}


-- Create an event tap to intercept key down events
cmdHEventTap = hs.eventtap.new({hs.eventtap.event.types.keyDown}, function(event)



  local mods = event:getFlags()
  local keycode = event:getKeyCode()

  -- Check if it's precisely Cmd+H (only Command modifier down, 'h' key)
  if mods:containExactly({"cmd"}) and keycode == hs.keycodes.map.h then
    local frontApp = hs.application.frontmostApplication()

    -- Ensure we got the frontmost application object
    if frontApp then
      local frontAppID = frontApp:bundleID()

      -- Check if the frontmost app's bundle ID is in our VS Code list
      if vscodeBundleIDs[frontAppID] then
        -- It is VS Code! Tell the application object to hide.
        frontApp:hide()
        -- Return true to signify we've handled the event and it should not be passed further.
        -- This prevents the key event from reaching VS Code's webview.
        return true
      end
    end
  end

  -- If it wasn't Cmd+H for VS Code, or if something went wrong,
  -- return false to let the event propagate normally.
  return false
end)

-- Start the event tap so it starts listening for key events
cmdHEventTap:start()

-- Optional Cleanup: Stop the tap if Hammerspoon quits or reloads its config
-- This prevents potential zombie taps if you modify the script significantly later.
-- Usually handled automatically when the variable is reassigned on reload, but good practice.
hs.shutdownCallback = function()
    if cmdHEventTap then
        cmdHEventTap:stop()
        cmdHEventTap = nil
    end
end

--[[
END of Workaround for VS Code Markdown Preview intercepting Cmd+H.
--]]
```

**Comment by FunProgramer** [comment]
I have the same problem but with shortcuts defined using the macOS Shortcuts app.

**Comment by codydh** [comment]
Same here, this popped up over the past few weeks.
